snowflake_blueprints/upload_file.py

```diff
@@ -65,23 +59,209 @@ def get_args():
     return args
 
 
-def upload_data(source_full_path, table_name, insert_method, db_connection):
+def create_table(source_full_path, table_name, insert_method, db_connection):
+    """
+    Creates a table by looking at the schema of the first 10k rows and only loading the header row.
+    Used by the new PUT method because you can't PUT or COPY INTO if the table doesn't exist beforehand.
+    """
     try:
+        chunksize = 10000
         for index, chunk in enumerate(
-                pd.read_csv(source_full_path, chunksize=10000)):
-
-            if insert_method == 'replace' and index > 0:
-                # First chunk replaces the table, the following chunks
-                # append to the end.
-                insert_method = 'append'
-
-            chunk.to_sql(
+                pd.read_csv(source_full_path, chunksize=chunksize)):
```

Comment:
What is `source_full_path` and what is `chunk.head(0).to_sql()`? 
You shouldn't need a chunk of 10000 is all you are doing is loading a row to determine the table table and then creating a table. Chunk size of 1 would work. 

Does snowflake support creating a table like another table so you don't have to read and use pandas to determine the schema?

Reply:
@ericelsken I'm eliminating hard work by relying on existing systems and ensuring parity with the existing upload behavior.

The existing `to_sql()` method we use automatically creates a table if one doesn't exist by looking at the schema type of the data, then it starts inserting data. Since we chunk the results, the existing method always created a table using the first 10k rows as a reference point.

The new PUT/COPY INTO method does not create a table and errors out if no table exists. Rather than building out custom "CREATE TABLE" logic that scans the schema of the provided CSV, I just relied on the existing logic to initially create the table. However, we don't want `to_sql()` to actually insert any data. `chunk.head(0)` ensures that only the first row (header) is added to the table but the entire chunk (10k rows) is loaded as a dataframe to determine the schema.

Reply:
>source_full_path
Combination of source_file_name and source_folder_name

Reply:
>Does snowflake support creating a table like another table so you don't have to read and use pandas to determine the schema?

They have a CREATE TABLE function, but this requires the schema to be specified up front. In most cases, we're creating a brand new table, not a table like another.